### PR TITLE
Feat/aggregation

### PR DESCRIPTION
**Pull request by Xuyen21**
adding data table , when user click on graph, this data table appear
adding title for graphs

### CODE REVIEW

**Comment by Xuyen21** [comment]
I used auto format code, so it showed a lot changes for the alignment in file postgres.py. Sorry for that

File: app/pages/aggres/aggres_rendering.py [hunk 1]
```diff
@@ -48,4 +53,34 @@ def draw_aggregation(selected_table: str = 'author', chart_type: str = 'bar char
...
+        """
+
+        current_value = click_data['points'][0]['x']  # return name of the author/school... from dropdown
+        total_publications = click_data['points'][0]['y']  # return number of publications
+        # get data according to the chosen name
+        data_info = get_publications_table(current_value)
+        result = dash_table.DataTable(data_info,
+                                      style_data={'height': 'auto', 'whiteSpace': 'normal', 'textAlign': 'left',
+                                                  'padding': '5px'}, style_header={'textAlign': 'center'})
+        overview_discription = f'{current_value} has total of {total_publications} publications'
```
Comment (SilvanW) [inline app/pages/aggres/aggres_rendering.py]: f'{current_value} has **a** total of {total_publications} publications'

File: app/modules/postgres.py [hunk 2]
```diff
@@ -134,12 +138,12 @@ def author_relations(table: str, limit: int) -> list[tuple]:
         ORDER BY COUNT(entry_author.author_id) DESC
         LIMIT {limit}
     ) as sub_col"""
-    ).format(
-        select = sql.SQL(f'{table}.entry_key'),
-        table = sql.Identifier(table),
-        group_by = sql.SQL(f'{table}.entry_key'),
-        on = sql.SQL(f'{table}.entry_key'),
-        limit = sql.Literal(limit)
+                        ).format(
```
Comment (SilvanW) [inline app/modules/postgres.py:141]: What a weird format

File: app/pages/aggres/aggres_rendering.py [hunk 3]
```diff
@@ -34,12 +35,16 @@ def draw_aggregation(selected_table: str = 'author', chart_type: str = 'bar char
         df = pd.DataFrame(values, columns=selected_columns)
         df = df.sort_values(by=[count], ascending=True)
 
+        chart_title = f'The {selected_table}s in top {data_limit} publications'
```
Comment (SilvanW) [inline app/pages/aggres/aggres_rendering.py]: When selecting the pages it says: pagess

Comment (SilvanW) [reply app/pages/aggres/aggres_rendering.py]: I would just call it: The top {data_limit} {selected_table}

**Review by SilvanW — CHANGES_REQUESTED** [review]
Has a few issues with text - otherwise it's good

File: app/modules/postgres.py (hunk 2 above)
Comment (Xuyen21) [reply app/modules/postgres.py:141]: Haha, maybe sometimes it involves also the manual format from me, and yes, it's weird :)

File: app/pages/aggres/aggres_rendering.py (hunk 1 above)
Comment (Xuyen21) [reply app/pages/aggres/aggres_rendering.py]: ok

File: app/pages/aggres/aggres_rendering.py (hunk 3 above)
Comment (Xuyen21) [reply app/pages/aggres/aggres_rendering.py]: ok

**Review by SilvanW — APPROVED** [review]

**Review by KarolinaT92 — APPROVED** [review]

**Pull request merged**